core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java

```diff
@@ -12147,6 +12153,11 @@ private void clearUserIDResolverCache(String userID, String userName, UserStore
                 .clearCacheEntry(UserCoreUtil.addDomainToName(userName, userStore.getDomainName()),
                         RESOLVE_USER_ID_FROM_USER_NAME_CACHE_NAME, tenantId);
         UserIdResolverCache.getInstance().clearCacheEntry(userID, RESOLVE_USER_NAME_FROM_USER_ID_CACHE_NAME, tenantId);
+        UserIdResolverCache.getInstance()
+                .clearCacheEntry(UserCoreUtil.addDomainToName(userName, userStore.getDomainName()),
```

Comment:
create a issue to track cache clear in user name update (will be supported in future)

Reply:
create an issue to replace usage of `LIKE` operator used method for getUser method use case

Reply:
https://github.com/wso2/product-is/issues/9322

Reply:
https://github.com/wso2/product-is/issues/9323